modules/cli/src/main/scala/scala/cli/commands/Bsp.scala

```diff
@@ -44,8 +44,8 @@ object Bsp extends ScalaCommand[BspOptions] {
             pprint.err.log(initialInputs)
 
           val buildOptions0    = buildOptions(sharedOptions)
-          val logger           = sharedOptions.logging.logger
-          val persistentLogger = new PersistentDiagnosticLogger(logger)
+          val latestLogger     = sharedOptions.logging.logger
+          val persistentLogger = new PersistentDiagnosticLogger(latestLogger)
```

Comment:
Because `BSP` refreshes its options based on the latest connection details.
The logger coming from the latest `SharedOptions` instance is used here in case i.e. the verbosity was changed since the last `BSP` `workspace/reload`.